src/math/dd/dd_bdd.h

```diff
@@ -303,17 +299,102 @@ namespace dd {
 
-        /** Returns an integer contained in the BDD, if any, and whether the BDD is a singleton. */
-        // NSB code review: this API needs to be changed: bit-position to variable mapping is external
-        find_int_t find_int(unsigned w, rational& val) { return m->find_int(root, w, val); }
+        /** Returns an integer contained in the BDD, if any, and whether the BDD is a singleton.
+         *
+         * Preconditions:
+         * - bits are sorted in ascending order,
+         * - the bdd only contains variables from bits.
+         */
+        find_result find_num(unsigned_vector const& bits, rational& val) const { return m->find_num(root, bits, val); }
     };
 
     std::ostream& operator<<(std::ostream& out, bdd const& b);
 
+    class bddv {
+        friend bdd_manager;
+
+        // TODO: currently we repeat the pointer to the manager in every entry of bits.
+        // Should use BDDV instead (drawback: the functions in bdd_manager aren't as nice and they need to be careful about reference counting if they work with BDD directly).
+        vector<bdd>  bits;
+        bdd_manager& m;
+
+        bddv(vector<bdd> bits, bdd_manager* m): bits(bits), m(*m) { inc_bits_ref(); }
+        bddv(vector<bdd>&& bits, bdd_manager* m): bits(std::move(bits)), m(*m) { inc_bits_ref(); }
+        void inc_bits_ref() {
+            // NOTE: necessary if we switch to BDDV as storage
+            // for (BDD b : bits) { m.inc_ref(b); }
+        }
+
+        // NOTE: these should probably be removed if we switch to BDDV
+        bddv(bdd_manager* m): bits(), m(*m) { }
+        bdd const& operator[](unsigned i) const { return bits[i]; }
+        bdd& operator[](unsigned i) { return bits[i]; }
+        void push_back(bdd const& a) { bits.push_back(a); }
+        void push_back(bdd&& a) { bits.push_back(a); }
+    public:
+        bddv(bddv const& other): bits(other.bits), m(other.m) { inc_bits_ref(); }
+        bddv(bddv&& other): bits(), m(other.m) { std::swap(bits, other.bits); }
+        bddv& operator=(bddv const& other) {
+            if (this != &other) {
+                bddv old(std::move(*this));
+                bits = other.bits;
+                inc_bits_ref();
+            }
+            return *this;
+        }
+        bddv& operator=(bddv&& other) {
+            if (this != &other) {
+                bddv old(std::move(*this));
+                SASSERT(bits.empty());
+                std::swap(bits, other.bits);
+            }
+            return *this;
+        }
+        ~bddv() {
+            // NOTE: necessary if we switch to BDDV as storage
+            // for (BDD b : bits) { m.dec_ref(b); }
+        }
+
+        unsigned size() const { return bits.size(); }
+        vector<bdd> const& get_bits() const { return bits; }
+
+        bdd operator<=(bddv const& other) const { return m.mk_ule(*this, other); }
```

Comment:
Just expose them as sle, ule, sgt, uge, ... instead of opeators. We don't have sign information in the bits.

Reply:
Or keep these for niceness of unit tests, but add signed versions using sle, sgt.